plone/app/content/namechooser.py

```diff
@@ -45,7 +45,7 @@ def chooseName(self, name, obj):
                 name = obj.__class__.__name__
 
         if not isinstance(name, six.text_type):
-            name = six.text_type(name, 'utf-8')
+            name = name.encode('utf-8')
```

Comment:
This looks wrong. What we want is to make sure that the name is text, but this code tries to encode it if it is NOT text, which makes no sense.